inputs/costs/costs_biomass.ad

```diff
@@ -1,13 +1,14 @@
 # This now updates ALL biomass products, even woody_biomass (but not bio-fuels). This is required as fix to prevent negative costs from the export of wood_pellets. When a flex_max link type is made possible, we should rethink the workings of this slider
-
-- query = UPDATE(CARRIER(greengas,biogas,wood_pellets,woody_biomass,wood,torrified_biomass_pellets), cost_per_mj, USER_INPUT())
-- label = &euro;/tonne
-- label_query = Q(price_of_wood_pellets)
+- query =
+  present_value = QUERY_PRESENT(-> {Q(price_of_wood_pellets)});
+  relative_value = USER_INPUT() / present_value;
```

Comment:
I also created [variables](https://github.com/quintel/etsource/pull/1500#issue-284126917) in a query once, yet refrained from doing so. I'm still not sure where the balance in this case lies, maybe @antw has a stronger opinion in this case. 

One side-note: what is the difference between: 

```
QUERY_PRESENT(-> {Q(price_of_wood_pellets)});
```

and

```
present:Q(price_of_wood_pellets)
```

Doesn't this return the same thing?

Reply:
I first tried ``present:Q(price_of_wood_pellets)`` but it didn't work

Reply:
> One side-note: what is the difference between: `QUERY_PRESENT` and `present:`

`present:` is used as a prefix for the entire query, not a single expression.

For example:

**Yes:**

```ruby
- start_value_gql = present:Q(price_of_wood_pellets)
```

**No:**

```ruby
- start_value_gql =
    value = present:Q(price_of_wood_pellets) # ← syntax error
    value * 10
```

---

Nonetheless, I think...

```ruby
QUERY_PRESENT(-> {Q(price_of_wood_pellets)})
```

... can be simplified as ...

```ruby
QUERY_PRESENT(price_of_wood_pellets)
```

Reply:
> I also created variables in a query once, yet refrained from doing so. I'm still not sure where the balance in this case lies, maybe @antw has a stronger opinion in this case.

This is personal opinion, but I would refrain from using variables unless they make a complex query much easier to read and understand. Variables are a quirk of GQL being implemented in Ruby rather than something which is officially supported. However, I doubt support for variables will ever be removed, so... */shrug*.

Here's how I would write the query:

```ruby
UPDATE_WITH_FACTOR(
  CARRIER(
    biogas,
    greengas,
    torrified_biomass_pellets,
    wood,
    woody_biomass,
    wood_pellets
  ), 
  cost_per_mj,
  USER_INPUT() / QUERY_PRESENT(price_of_wood_pellets)
)
```